### ISSUE TITLE
🐞 Bug: The search feature should be a little more flexible and user-friendly. 

### ISSUE THREAD

**Issue by Chandrachur67** [CircuitVerse/CircuitVerse#4782]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

The main search option in CircuitVerse is very strict. Making it a little bit flexible might help users find things easily. 

Examples: 
If we try to search user ```"chandrachur"```, our input should be exactly ```"chandrachur"``` and seaching ```"chandrachu"``` won't give any results. 
 
Similar is the case for projects also. 
Searching ```"big project"``` gives many results, but ```"big projec"``` gives no results. 

### Steps To Reproduce

1. Go to https://circuitverse.org/
2. Search chandrachur and chandrachu and see the results.
3. Try with some other projects or users also

### Expected Behavior

This behaviour should be probably little more flexible. maybe we should implement a better searching algo to get better results.

### Screenshots

| ![Screenshot 1](https://github.com/CircuitVerse/CircuitVerse/assets/97349505/be412460-c8af-4920-b12d-06690b3f7e10) | ![Screenshot 2](https://github.com/CircuitVerse/CircuitVerse/assets/97349505/7ae123e5-5be7-4662-aa3f-1431147993cf) |
|-------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------|
| ![Screenshot 3](https://github.com/CircuitVerse/CircuitVerse/assets/97349505/6777c637-1cfd-4d8e-8d6d-70346a514ed2) | ![Screenshot 4](https://github.com/CircuitVerse/CircuitVerse/assets/97349505/6bd2e499-9a62-4bd5-96e7-7366b305e260) |

### Device Information [optional]

```markdown
- OS: MacOS
- Browser: 
- version:
```

### Additional context

N/A

### Are you working on this issue?

Yes

**Comment by Chandrachur67** [comment]
Hey @tanmoysrt I would like to work on this issue. 

Also, I would like to know about your suggestions to improve the search feature.

**Comment by Chandrachur67** [comment]
@Malavi1 yes got it, thanks. But seems like tinkercad also has very strict matching restrictions. We should probably improve that also. @vedant-jain03 WDYT?

**Comment by ChandrakantaMandal** [comment]
Hi maintainers 👋
I’m interested in working on this issue and would love to contribute. If the issue is still open, please assign it to me.
